modules/redhen_connection/src/Plugin/Action/ChangeConnectionRole.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Drupal\redhen_connection\Plugin\Action;
+
+/**
+ * Change the role of a Redhen Connection.
+ *
+ * @Action(
+ *   id = "connection_change_role_action",
+ *   label = @Translation("Change the role for the selected connection(s)"),
+ *   type = "redhen_connection"
+ * )
+ */
+class ChangeConnectionRole extends ChangeConnectionRoleBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($connection = NULL) {
+    $role = $this->configuration['role'];
+    $mismatch = FALSE;
+    // Skip changing the role for the connection if already present,
+    // OR if the role is not available for the given connection.
+    $entity_storage = \Drupal::service('entity_type.manager')->getStorage('redhen_connection_role');
```

Comment:
`$entity_storage` is sort of a confusing name here. `$role_storage` might be clearer, if I'm reading this right?

Reply:
yes, good call!